src/web/shared/component/Form.tsx

```diff
@@ -38,7 +39,7 @@ const Form = ({
   const [searchInput, setSearchInput] = useState('');
   const [selectedJuz, setSelectedJuz] = useState(undefined);
   const [selectedSurah, setSelectedSurah] = useState(undefined);
-  const [selectedApproach, setSelectedApproach] = useState(ApproachOptions()[0]);
+  const [selectedApproach, setSelectedApproach] = useState(approachOptions()[0]);
```

Comment:
When setting an initial state that requires heavy calculations, be sure to use lazy initialization.
This way, the calculation is performed only once, during the initial render or when it's absolutely necessary.

```suggestion
  const [selectedApproach, setSelectedApproach] = useState(() => approachOptions()[0]);
```

Reply:
updated in https://github.com/RR-Ardhika/murojaah/pull/35/commits/b2171edc38efca4af40e24d6ed04a45e9dfd6276, thank you